### DISCUSSION TITLE
Endpoints

### DISCUSSION THREAD

**Discussion by Alicedale101** [cloudbet/docs#68] [General]
Good day, is there a endpoint that allows me to retrieve the odds of the games? busy building a bot that retrieves the odds of all the games and then places bets on the but i keep running into a 404 error with the fetching the odds endpoint

**Comment by kgravenreuth** [comment]
Hi @Alicedale101 

You can query for a list of all events and competitions for a given sport. There are several sample articles on how to do this, for example:

1. [How to build a betting bot with the Cloudbet API](https://bitedge.com/blog/how-to-build-a-betting-bot-with-the-cloudbet-api/) -> This also shows you how to bet when specific odds conditions are met.
2. [Sports Betting API Tutorial With Golang](https://www.cloudbet.com/en/blog/posts/sports-betting-api-tutorial-with-golang)

For example, if you want to follow a specific team, then you can use our `/v2/odds/events` or `/v2/odds/fixtures` endpoint for a specific sport and then parse the response. The `home` and `away` fields of the response list the team names and keys which can be used by your bot to follow specific teams.

Please let us know if this meets your requirements. 

For the 404 response, could you please post a curl example of the request, so that we could explain the reason for why it might be giving a 404?

Regards,
Klaus Gravenreuth
Product & Engineering @ Cloudbet

**Comment by Alicedale101** [comment]
I can see the endpoints to get the odds for the soccer matches but wanted to know if it is possible to say the get all the live odds from all the games/sport that is live at that moment to say if you have a bot to checks all the odds constantly that will place bets on all sport/games that has a odd of 1.2 for example

**Comment by Alicedale101** [comment]
Hi Klaus, thank you for the reply and all your patience. that's all i
wanted to know if it is possible to create a bot that actually keeps
retrieving the odds of say all tennis games or rugby and when the odds are
met with the live games not with odds before the game starts but while the
game is on to place a bet on one game if the odds are met. but it seems it
is possible. I just need to brush up on my python skills. but thank you
again.
